### PR TITLE
[NEW] Route to get updated roles after a date

### LINKED ISSUE #16599: Route to get updated and deleted roles after a date
We only have a REST API to list roles: https://github.com/RocketChat/Rocket.Chat/blob/develop/app/api/server/v1/roles.js#L8

We should have another route to fetch `updated` and `deleted` roles after a date, so it'll be easier to maintain a local database on mobile.

### PR DESCRIPTION
**Pull request by ashwaniYDV**
Closes #16599

### CODE REVIEW

File: app/api/server/v1/roles.js [hunk 1]
```diff
@@ -13,6 +13,19 @@ API.v1.addRoute('roles.list', { authRequired: true }, {
 	},
 });
 
+// api to to get updated roles after a date(in ISODate format)
+API.v1.addRoute('roles.listByUpdatedDate', { authRequired: true }, {
+	get() {
+		check(this.bodyParams, {
+			updatedAfter: String,
```
Comment (MarcosSpessatto) [inline app/api/server/v1/roles.js]: @ashwaniYDV can you please validate this query parameter, to make sure it is a valid date? Here we have an example: https://github.com/RocketChat/Rocket.Chat/blob/develop/app/api/server/v1/rooms.js#L34-L43

File: app/api/server/v1/roles.js [hunk 2]
```diff
@@ -13,6 +13,19 @@ API.v1.addRoute('roles.list', { authRequired: true }, {
...
+		});
+
+		const roles = Roles.find({ _updatedAt: { $gte: new Date(this.bodyParams.updatedAfter) } }).fetch();
```
Comment (MarcosSpessatto) [inline app/api/server/v1/roles.js]: Can you create a Model method and put this logic inside it to avoid model logic leak?

**Comment by ashwaniYDV** [comment]
@MarcosSpessatto I have done the requested changes. Please review

File: app/api/server/v1/roles.js (hunk 2 above)
Comment (ashwaniYDV) [reply app/api/server/v1/roles.js]: Yes sure. I'll do it :)

File: app/api/server/v1/roles.js (hunk 1 above)
Comment (ashwaniYDV) [reply app/api/server/v1/roles.js]: Done :+1:

File: app/models/server/models/Roles.js [hunk 3]
```diff
@@ -85,6 +85,14 @@ export class Roles extends Base {
 		return this.findOne(query, options);
 	}
 
+	findByUpdatedDate(updatedAfterDate, options) {
+		const query = {
+			_updatedAt: { $gte: new Date(updatedAfterDate) },
+		};
+
+		return this.find(query, options).fetch();
```
Comment (MarcosSpessatto) [inline app/models/server/models/Roles.js]: I think you can put this `fetch` outside the method(use when you call the `findByUpdatedDate`), this way we can call another model methods like `count`, `forEach`, etc

**Comment by MarcosSpessatto** [comment]
@ashwaniYDV can you please add some tests?

**Comment by ashwaniYDV** [comment]
@MarcosSpessatto I have merged the two functionalities in the same function keeping it consistent to the other apis
I have added the checks too.

**Comment by MarcosSpessatto** [comment]
@ashwaniYDV I did some adjustments like keep the two endpoints. Can you please add docs for the new endpoint?

**Comment by ashwaniYDV** [comment]
hi @MarcosSpessatto 
You mean this repo https://github.com/rocketchat/docs

**Pull request merged**